### ISSUE TITLE
Ionic 2 version request.

### ISSUE THREAD

**Issue by rohamak** [katemihalikova/ion-datetime-picker#107]
Hello, This is a great control, can you make an ionic 2 version of this ? I want to use for desktops since ion-datatime-picker for ionic 2 iOS like is unusable and very difficult to use on desktops specially when using track pad it is very difficult to go from beginning to ends and vice versa.
Thanks.

**Comment by katemihalikova** [comment]
Moving from #104:

> @katemihalikova commented 10 hours ago
> I'm currently working on a new version of the picker that will work with both v1 and v2, but it's still not finished. However I can quickly convert current code (just simplified version of it) to v2 directive if you are interested in a quick solution.

> @rohamak commented 3 hours ago
> I looked at it, I think there is an injectable needed the i18n .factory and a ModalControlView to replace showPopup since the popup class does not exist in ionic 2 the way to pass templateUrl and cssClass, and a bunch ionic directive changes in the picker-popup.html.
That's my guesses so there might be some other stuff as well.

**Comment by katemihalikova** [comment]
I'll hook it into existing `ion-datetime` so that the api is almost the same. Stay tuned 📶

**Comment by katemihalikova** [comment]
Sneak-peek

<img src="https://cloud.githubusercontent.com/assets/7225023/25885039/7da9f1a4-3556-11e7-9948-864601c65af3.png" width="47%" align="left">
<img src="https://cloud.githubusercontent.com/assets/7225023/25885067/a1702360-3556-11e7-930f-e847d94b03de.png" width="47%" align="left">

**Comment by rohamak** [comment]
wow this looks fantastic, looking forward to using it. great job.

**Comment by katemihalikova** [comment]
Would you please give it a try? [Readme](https://github.com/katemihalikova/ion-datetime-picker/tree/ionic3)

```css
npm install ion-datetime-picker@ionic3
```

**Comment by rohamak** [comment]
thanks for the update however I get an error which I'm not sure if I should set some thing correctly.
I followed your instructions.

<img width="704" alt="screen shot 2017-05-10 at 8 42 34 pm" src="https://cloud.githubusercontent.com/assets/1694511/25927124/8b6bfb5a-35c1-11e7-979c-4886693db77c.png">

**Comment by katemihalikova** [comment]
Thanks for trying, I hope you don't mind a little debugging ☺ Could you please share your `app.module.ts`? The error you posted usually means that the module is under `declarations`, not under `imports` where it belongs. Also please make sure that the ionic version you are using is >=3.0.1.

**Comment by rohamak** [comment]
Sure no problem, happy to participate, that's right I overlooked that, I changed it to imports and it works, but there are a couple of issues, 

- first when ion-label and ion-datetime picker are under the same ion-col the generated DOM is not clickable, then I put them in two different ion-col s but this way user has to click on the blank area in from of the date presentation, instead of clicking on the date itself.
Maybe ion-date picker should be wrapped in a button directive ?
<img width="381" alt="issue 1" src="https://cloud.githubusercontent.com/assets/1694511/25939620/7707935a-3601-11e7-9b52-5c0f9dd28bec.png">
<img width="561" alt="issue 1" src="https://cloud.githubusercontent.com/assets/1694511/25939621/7716764a-3601-11e7-8e0d-6f567e22c89b.png">

- When the picker shows up input boxes are shrunk vertically, I attached pictures.
<img width="310" alt="issue 2" src="https://cloud.githubusercontent.com/assets/1694511/25939622/7718587a-3601-11e7-9ae9-c027ecafeab8.png">
Hope it helps. thank you.

**Comment by katemihalikova** [comment]
- ` ion-label` and `ion-datetime` should be both put inside `ion-item`, otherwise the behaviour is weird even with the original unmodified version of the picker.
- what browser and operating system are you using for testing this? I'd like to reproduce.

**Comment by rohamak** [comment]
yes, sure, this is a great work done is a short amount of time, I can take care of these issues by downloading the project from github , so if you want to spend more time on it for the purpose of public, other wise I'm more than fine with what is already here.
here are my settings : 

<img width="385" alt="screen shot 2017-05-11 at 10 09 52 am" src="https://cloud.githubusercontent.com/assets/1694511/25954203/89766c32-3633-11e7-8dce-f43e90469113.png">
<img width="287" alt="screen shot 2017-05-11 at 10 10 26 am" src="https://cloud.githubusercontent.com/assets/1694511/25954204/8991ac68-3633-11e7-8b6a-4e1f4d4e3d90.png">
<img width="246" alt="screen shot 2017-05-11 at 10 10 53 am" src="https://cloud.githubusercontent.com/assets/1694511/25954205/899e92d4-3633-11e7-8716-4a1290d03898.png">

I have changed it to the following but still label is not clickable, which I think I can do some thing about it so don;t worry a bout that.

<img width="562" alt="screen shot 2017-05-11 at 10 26 29 am" src="https://cloud.githubusercontent.com/assets/1694511/25954494/855db654-3634-11e7-8597-65c5c6a88523.png">

**Comment by katemihalikova** [comment]
I've found some old Mac with Safari 9.1 and I'm able to reproduce, thank you very much. It seems that Safari is very picky about some styles, I'll try to fix them soon.

**Comment by rohamak** [comment]
That's good to be able see the problem, and hope it's not going to take a lot of your time. :)

**Comment by claudiozam** [comment]
Hi, when the picker shows up input boxes are shrunk vertically. Tested on IPhone 5S - 10.3
Ionic 3.

**Comment by katemihalikova** [comment]
Yep, they are. I currently don't have much time to dig through Safari issues, but feel free to open a PR 👍

**Comment by betatools** [comment]
A quick dirty fix for me was to change two lines in picker.component.js file.

Line 321: `<div class="item item-input item-select" style="min-height: 25px;">`
Line 330: `<input type="number" [(ngModel)]="bind.year" min="1900" max="2999" (change)="change('year')" (blur)="changed()" required style="min-height: 25px;">`

Just added the two style attributes

**Comment by jbarbede** [comment]
hi @katemihalikova, I'd like to help to improve that component for last ionic version but first I face some AOT compilation error: Error: Unexpected value 'DatetimePickerModule in 
            .../node_modules/ion-datetime-picker/dist/picker.module.d.ts' imported by the 
            module 'AppModule in .../src/app/app.module.ts'. Please add a @NgModule 
            annotation.

I will check what is happening deeper this week-end but if you have any idea, let me know what could cause that.

**Comment by node-ist** [comment]
The same problem as @jbarbede  has described - error during AOT compilation

**Comment by jbarbede** [comment]
I sent a PR to make the module AOT compatible. Please take a look @katemihalikova. I need your comments, specially on some fix I had to make on the picker directive.

**Comment by roggerlacerda** [comment]
Hi, @katemihalikova! Could you update the peerDependencies to newer ionic versions, please? I'm facing an error "angular is not defined" when using the component. Thanks!

**Comment by husainsr** [comment]
I faced same problem AOT compilation error: Error: Unexpected value 'DatetimePickerModule in
.../node_modules/ion-datetime-picker/dist/picker.module.d.ts' imported by the
module 'AppModule in .../src/app/app.module.ts'. Please add a @NgModule
annotation.


what should i do ?

**Comment by mubasshir** [comment]
hi, any update on this? 
How to resolve this?

**Comment by katemihalikova** [comment]
You may try using alpha-1 using `npm install @ion-datetime-picker/core@1.0.0-alpha.1 --save-exact`.

**Comment by mydoal** [comment]
`npm install @ion-datetime-picker/core@1.0.0-alpha.1 --save-exact`
fixes the AOT compilation problems. Thanks.
